### ISSUE TITLE
QFT Question

### ISSUE THREAD

**Issue by abrassel** [microsoft/QuantumKatas#817]
Hi!  Thanks for these incredible tutorials.  I'm reading the QFT Kata, and trying to understand the step from j/2^k -> 0.j_n...j_n-k in Task 1.7  Would anyone be able to elaborate on the arithmetic there?

**Comment by Manvi-Agrawal** [comment]
Hi @abrassel, I did not get your question fully,  so I  am taking liberty to explain the whole solution.

- First, we start by preparing the described state with the qubits reversed. This is important as it allows to prepare the state inplace. 
- We are doing this by leveraging `BinaryFractionQuantumInPlace` from Task 1.5. This function changes the state of the register from $|j_1\\rangle \\otimes |j_2 ... j_n\\rangle$ to $\\frac{1}{\\sqrt{2}} \\big(|0\\rangle + e^{2\\pi i \\cdot 0.j_1 j_2 ... j_n} |1\\rangle \\big) \\otimes |j_2 ... j_n\\rangle$. 
- In short, `BinaryFractionQuantumInPlace` if applied on register $|j_1 ... j_n\\rangle$, only transforms $|j_1\\rangle$.
- We can use above operation to transform $|j_2>$ to $\\frac{1}{\\sqrt{2}} \\big(|0\\rangle + e^{2\\pi i \\cdot 0.j_2 ... j_n}$ by applying operation `BinaryFractionQuantumInPlace` to register $|j_2 ... j_n\\rangle$, and so on...
- Finally, we use `ReverseRegister` from task 1.6 to get the desired state.

**Comment by abrassel** [comment]
Hi @Manvi-Agrawal , thanks for explaining your solution!  I didn't realize that you could use latex in markdown comments, so I can actually go into a bit more depth about the nature of my question.  My specific understanding gap is in the math, not the gates we are using to implement it.

In Task 1.7, there is a two-step sequence of algebraic transformations.  From $\frac{1}{\sqrt{2^n}}\sum_{k=0}^{2^n-1}{e^{2\pi i \cdot \frac{jk}{2^n} |k\rangle$, and taking any given qubit (i.e. $|0101\rangle$), and factoring it across tensor products.  When you distribute the tensor products, you wind up with terms such as $e^{2\pi i \cdot (0.j_3j_2 + 0.j_1j_2_j3)}|011\rangle$.  It's unclear to me how that is equivalent to the step prior to the transformation.

**Comment by Manvi-Agrawal** [comment]
Hi @abrassel, the math of transformation from $\frac{1}{\sqrt{2^{n}}} \sum_{}{} e^{2\pi i \cdot \frac{jk}{2^{n}}} |k\rangle$ to factoring it across various tensor products is a bit intensive and lengthy.

Here is the [binder link](https://mybinder.org/v2/gh/Manvi-Agrawal/QuantumKatas/qft/t17-qft-math?labpath=QFT%2FWorkbook_QFT.ipynb#Math-to-get-qubit-transformation-from-summation) that you can visit to see the math.

**Comment by Manvi-Agrawal** [comment]
@tcNickolas, do you think we should add this math in workbook or would it get too intensive? Because I think its not very obvious to understand the transformation from summation to tensor products expression listed in workbook for task 1.7.

**Comment by abrassel** [comment]
Oh!  This helped immensely @Manvi-Agrawal .  I would definitely have benefited from having this in the work book (or at least a link to the solutions).

**Comment by Manvi-Agrawal** [comment]
I am glad @abrassel , that I was able to help you

**Comment by tcNickolas** [comment]
Yes, if this helps, we definitely can add this to the workbook! (And I didn't know you could use LaTeX on GitHub either! :-) )

**Comment by Manvi-Agrawal** [comment]
Hmm, this is interesting. This was my first time using LaTex on Github. My idea behind using LaTex in comment was that Github supports markdown in comments. And we use Latex in markdown cells in Jupyter notebook. That being said, Github markdown supports only some LaTex or with a slightly different syntax. 

I initially planned to send the math explaination in Github comment by copying from jupyter notebook. But it did not render well. So, I just commited my changes in jupyter and shared Github llink to save time.

**Comment by Manvi-Agrawal** [comment]
I thought it might be a good idea to introduce a separate task for `Two qubit QFT` so that math for `N bit QFT` does not sound a bit lengthy to the learners.

 Also, I thought that maybe we should give the info about transformation of summation to tensor products in a hint, rather than giving it right away so that learners can spend some time in figuring out math. Also, I have given an option to skip math as well, so that if someone is finding math difficult they can just try to implement the expression instead of leaving the kata unsolved.
 
Here is my [link](https://github.com/Manvi-Agrawal/QuantumKatas/commits/qft-tasks-refactor) to my branch containing the changes. I have 3 commits. Kindly note that [first commit](https://github.com/Manvi-Agrawal/QuantumKatas/commit/b76dddd917569ae0dfc77a4f3652b32894313563) contains most of the meaningful changes. Second commit is for cleaning up things, fixing typos and adding some stuff that I thought might be useful. Third commit just changes the task numbering.

Would love to know if this refactoring makes sense. Or do we just want to add math explaination from the [github link](https://github.com/Manvi-Agrawal/QuantumKatas/blob/qft/t17-qft-math/QFT/Workbook_QFT.ipynb#Math-to-get-qubit-transformation-from-summation) that I shared earlier.
